### PR TITLE
DataBase Action bug fix

### PR DESCRIPTION
**Pull request by AmanPrasad43**
Thank you for your contribution.
Before submitting this PR, please make sure:
- [x] PR description and commit message should describe the changes done in this PR
- [x] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [x] No unwanted\commented\junk code is included
- [x] No new warning upon build solution
- [x] Code Summary\Comments are added to my code which explains what my code is doing
- [x] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Help Libray document is updated accordingly for Feature changes.

### CODE REVIEW

File: Ginger/GingerCoreNET/Database/DatabaseOperations.cs [hunk 1]
```diff
@@ -469,65 +471,57 @@ public List<string> GetTablesList(string Keyspace = null)
...
+                                rc.Add(tableName);
+                            }
+                            break;
+                        default:
+                            //Not implemented
+                            break;
                     }
-                    else
-                    {
+                        if (Database.DBType == eDBTypes.MSSQL || Database.DBType == eDBTypes.MySQL || Database.DBType == eDBTypes.MSAccess || Database.DBType == eDBTypes.DB2 || Database.DBType == eDBTypes.PostgreSQL)
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/Database/DatabaseOperations.cs:514]: handle this condition in switch defined above. You can do it using the Case without break.

File: Ginger/GingerCoreNET/Database/DatabaseOperations.cs [hunk 2]
```diff
@@ -469,65 +471,57 @@ public List<string> GetTablesList(string Keyspace = null)
...
-                                case eDBTypes.PostgreSQL:
-                                    tableName = (string)row[2];
-                                    break;
-                                default:
-                                    //not implemented
-                                    break;
-                            }
-
-                            rc.Add(tableName);
+                                tableName = (string)row[2];
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/Database/DatabaseOperations.cs:520]: directly add (string)row[2] to list without defining the string tablename. Replace it in all places

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/GingerCoreNET/Database/DatabaseOperations.cs [hunk 3]
```diff
@@ -469,65 +471,57 @@ public List<string> GetTablesList(string Keyspace = null)
...
+
+                        case eDBTypes.Oracle:
+                            string[] restr = new string[1];
+                            restr[0] = GetConnectedUsername();
+                            DataTable table = await oConn.GetSchemaAsync("Tables", restr);
+                            string tableName = "";
+                            foreach (DataRow row in table.Rows)
+                            {
+                                tableName = (string)row[1];
+                                rc.Add(tableName);
```
[the remark this answers is not in the document]
Comment (AmanPrasad43) [reply Ginger/GingerCoreNET/Database/DatabaseOperations.cs:507]: Fixed

File: Ginger/GingerCoreNET/Database/DatabaseOperations.cs (hunk 1 above)
Comment (AmanPrasad43) [reply Ginger/GingerCoreNET/Database/DatabaseOperations.cs:514]: fixed

File: Ginger/GingerCoreNET/Database/DatabaseOperations.cs (hunk 2 above)
Comment (AmanPrasad43) [reply Ginger/GingerCoreNET/Database/DatabaseOperations.cs:520]: fixed

File: Ginger/GingerCoreNET/Database/DatabaseOperations.cs [hunk 4]
```diff
@@ -538,7 +532,17 @@ public List<string> GetTablesList(string Keyspace = null)
             return rc;
         }
 
-
+        /// <summary>
+        /// This function is for Oracle only
```
[the remark this answers is not in the document]
Comment (AmanPrasad43) [reply Ginger/GingerCoreNET/Database/DatabaseOperations.cs:536]: fixed

**Pull request merged**